dask_kubernetes/core.py

```diff
@@ -508,6 +508,35 @@ def f(to_close):
     def scale_up(self, n, **kwargs):
         self.scheduler.loop.add_callback(self._scale_up, n, **kwargs)
 
+    async def _pod_status(self, pods):
+
+        status = {}
+        for p in pods:
+            status[p.metadata.name] = p.status.phase
+        return status
+
+    async def check_status(self):
+        while True:
+            pods = await self._pods()
+            p_status = await self._pod_status(pods)
+            result = ['Running' == status for name, status in p_status.items()]
+            await asyncio.sleep(3.2)
+            if not all(result):
+                continue
```

Comment:
What if we scale down after scaling up within the timeout above?  Some of the pods might have a status of "done" or "cancelled" or something, right?